rustual-boy-core/src/v810.rs

```diff
@@ -908,11 +1021,51 @@ impl V810 {
             }
         }
 
+        trace_add!(TraceEntry::Exec(num_cycles));
+
         self.reg_pc = next_pc;
 
+        #[cfg(feature = "cpu-trace")]
+        self.trace_post_exec_data(interconnect);
+
         (num_cycles, trigger_watchpoint)
     }
 
+    #[cfg(feature = "cpu-trace")]
+    fn trace_post_exec_data(&mut self, interconnect: &mut Interconnect) {
+        if self.trace_enabled {
+            let mut post_entries: Vec<TraceEntry> = Vec::new();
+
+            macro_rules! trace_add_post {
```

Comment:
This macro seems a bit redundant, as this whole fn won't exist if the feature's not in, and all it's doing is pushing the entry to `post_entries`. Should probably just inline that code instead of making a macro for it.